spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/converter/DatasetProteomeEntryConverter.java

```diff
@@ -19,26 +30,338 @@
+    public static final String CONSORTIUM = "consortium";
+    public static final String ID = "_id";
+    public static final String ISOLATE = "isolate";
+    public static final String REDUNDANT_TO = "redundantTo";
+    public static final String PANPROTEOME = "panproteome";
+    public static final String REDUNDANT_PROTEOME = "redundantProteome";
+    public static final String EXCLUDED = "excluded";
+    public static final String SIMILARITY = "similarity";
+    public static final String EXCLUSION_REASON = "exclusionReason";
 
     @Override
     public ProteomeEntry call(Row row) throws Exception {
         ProteomeEntryBuilder builder = new ProteomeEntryBuilder();
-        builder.proteomeId(row.getString(row.fieldIndex("upid")));
-        // put upid and taxonomy from xml
+        // upid
+        builder.proteomeId(row.getString(row.fieldIndex(UPID)));
+        // taxonomy
         Taxonomy taxonomy =
-                new TaxonomyBuilder().taxonId(row.getInt(row.fieldIndex("taxonomy"))).build();
+                new TaxonomyBuilder().taxonId(row.getInt(row.fieldIndex(TAXONOMY))).build();
         builder.taxonomy(taxonomy);
-        // add if is a reference proteome
-        ProteomeType type = ProteomeType.NORMAL;
-        if (hasFieldName("isReferenceProteome", row)) {
-            boolean isReference = row.getBoolean(row.fieldIndex("isReferenceProteome"));
-            if (isReference) {
-                type = ProteomeType.REFERENCE;
-            }
-        }
-        builder.proteomeType(type);
+        // strain
+        if (hasFieldName(STRAIN, row)) {
+            builder.strain(row.getString(row.fieldIndex(STRAIN)));
+        }
+        // description
+        if (hasFieldName(DESCRIPTION, row)) {
+            builder.description(row.getString(row.fieldIndex(DESCRIPTION)));
+        }
+        // isolate
+        if (hasFieldName(ISOLATE, row)) {
+            builder.isolate(row.getString(row.fieldIndex(ISOLATE)));
+        }
+        // redundant to
+        if (hasFieldName(REDUNDANT_TO, row)) {
+            builder.redundantTo(
+                    new ProteomeIdBuilder((row.getString(row.fieldIndex(REDUNDANT_TO)))).build());
+        }
+        // panproteome
+        if (hasFieldName(PANPROTEOME, row)) {
+            builder.panproteome(
+                    new ProteomeIdBuilder((row.getString(row.fieldIndex(PANPROTEOME)))).build());
+        }
+        // modified
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        builder.modified(
+                LocalDate.parse(row.getDate(row.fieldIndex(MODIFIED)).toString(), formatter));
+        // proteome type
+        ProteomeType proteomeType = ProteomeType.NORMAL;
+        boolean isReference = row.getBoolean(row.fieldIndex(IS_REFERENCE_PROTEOME));
+        boolean isRepresentative = row.getBoolean(row.fieldIndex(IS_REPRESENTATIVE_PROTEOME));
+
+        if (isReference && isRepresentative) {
```

Comment:
Have a look in the ProteomeDocumentConverter about the logic that we use currently use to set these values, because we have other types of proteomes as well (exclude, ....)